src/aliceVision/camera/Pinhole.hpp

```diff
@@ -117,7 +117,15 @@ class Pinhole : public IntrinsicBase
     params.insert(params.end(), _distortionParams.begin(), _distortionParams.end());
     return params;
   }
-  
+
+  virtual bool hasDistortion() const
```

Comment:
it's better to use `override` instead of virtual when overriding a method. (there are more checks at compile time). The method will still be derived as `virtual` by default
https://en.cppreference.com/w/cpp/language/virtual